Mapstruct warning in master

default gradle build
```
{
  "generator-jhipster": {
    "jhipsterVersion": "2.27.0",
    "baseName": "demo",
    "packageName": "com.test.demo",
    "packageFolder": "com/test/sats",
    "authenticationType": "session",
    "hibernateCache": "ehcache",
    "clusteredHttpSession": "no",
    "websocket": "no",
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "mysql",
    "searchEngine": "elasticsearch",
    "buildTool": "gradle",
    "enableSocialSignIn": false,
    "rememberMeKey": "",
    "useSass": false,
    "enableTranslation": false,
    "applicationType": "monolith",
    "testFrameworks": []
  }
}
```
```
D:\Projects\demo\test\src\main\java\com\test\demo\web\rest\mapper\UserMapper.java:23: warning: Unmapped target properties: "createdBy, createdDate, lastModifiedBy, lastModifiedDate, id, activationKey, resetKey, resetDate, persistentTokens".
    User userDTOToUser(UserDTO userDTO);
         ^
1 warning
```

--- a/generators/server/templates/src/main/java/package/web/rest/mapper/_UserMapper.java
+++ b/generators/server/templates/src/main/java/package/web/rest/mapper/_UserMapper.java
@@ -18,16 +18,16 @@ public interface UserMapper {
     UserDTO userToUserDTO(User user);
 
     List<UserDTO> usersToUserDTOs(List<User> users);
-
+    <% if (databaseType != 'cassandra') { %>
     @Mapping(target = "createdBy", ignore = true)
     @Mapping(target = "createdDate", ignore = true)
     @Mapping(target = "lastModifiedBy", ignore = true)
-    @Mapping(target = "lastModifiedDate", ignore = true)
+    @Mapping(target = "lastModifiedDate", ignore = true)<% if (authenticationType == 'session') { %>
+    @Mapping(target = "persistentTokens", ignore = true)<% } } %>
     @Mapping(target = "id", ignore = true)
     @Mapping(target = "activationKey", ignore = true)
     @Mapping(target = "resetKey", ignore = true)
     @Mapping(target = "resetDate", ignore = true)
-    @Mapping(target = "persistentTokens", ignore = true)
     User userDTOToUser(UserDTO userDTO);
 
     List<User> userDTOsToUsers(List<UserDTO> userDTOs);

--- a/generators/server/templates/src/main/java/package/web/rest/mapper/_UserMapper.java
+++ b/generators/server/templates/src/main/java/package/web/rest/mapper/_UserMapper.java
@@ -22,7 +22,7 @@ public interface UserMapper {
     @Mapping(target = "createdBy", ignore = true)
     @Mapping(target = "createdDate", ignore = true)
     @Mapping(target = "lastModifiedBy", ignore = true)
-    @Mapping(target = "lastModifiedDate", ignore = true)<% if (authenticationType == 'session') { %>
+    @Mapping(target = "lastModifiedDate", ignore = true)<% if ((authenticationType == 'session') && (databaseType == 'sql')) { %>
     @Mapping(target = "persistentTokens", ignore = true)<% } } %>
     @Mapping(target = "id", ignore = true)
     @Mapping(target = "activationKey", ignore = true)

--- a/generators/server/templates/src/main/java/package/web/rest/mapper/_UserMapper.java
+++ b/generators/server/templates/src/main/java/package/web/rest/mapper/_UserMapper.java
@@ -1,14 +1,13 @@
 package <%=packageName%>.web.rest.mapper;
-
 <% if (databaseType == 'sql' || databaseType == 'mongodb') { %>
 import <%=packageName%>.domain.Authority;<% } %>
 import <%=packageName%>.domain.User;
 import <%=packageName%>.web.rest.dto.UserDTO;
-import org.mapstruct.Mapper;
+import org.mapstruct.*;
 
-import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
+import java.util.stream.Collectors;
 
 /**
  * Mapper for the entity User and its DTO UserDTO.
@@ -20,6 +19,15 @@ public interface UserMapper {
 
     List<UserDTO> usersToUserDTOs(List<User> users);
 
+    @Mapping(target = "createdBy", ignore = true)
+    @Mapping(target = "createdDate", ignore = true)
+    @Mapping(target = "lastModifiedBy", ignore = true)
+    @Mapping(target = "lastModifiedDate", ignore = true)
+    @Mapping(target = "id", ignore = true)
+    @Mapping(target = "activationKey", ignore = true)
+    @Mapping(target = "resetKey", ignore = true)
+    @Mapping(target = "resetDate", ignore = true)
+    @Mapping(target = "persistentTokens", ignore = true)
     User userDTOToUser(UserDTO userDTO);
 
     List<User> userDTOsToUsers(List<UserDTO> userDTOs);
@@ -33,21 +41,16 @@ public interface UserMapper {
         return user;
     }
 <% if (databaseType == 'sql' || databaseType == 'mongodb') { %>
-    default Set<String> stringsFromAuthorities(Set<Authority> authorities) {
-        Set<String> auths = new HashSet<>();
-        for(Authority auth : authorities){
-            auths.add(auth.getName());
-        }
-        return auths;
+        default Set<String> stringsFromAuthorities (Set < Authority > authorities) {
+        return authorities.stream().map(Authority::getName)
+            .collect(Collectors.toSet());
     }
 
     default Set<Authority> authoritiesFromStrings(Set<String> strings) {
-        Set<Authority> auths = new HashSet<>();
-        for(String auth : strings){
-            Authority a = new Authority();
-            a.setName(auth);
-            auths.add(a);
-        }
-        return auths;
+        return strings.stream().map(string -> {
+            Authority auth = new Authority();
+            auth.setName(string);
+            return auth;
+        }).collect(Collectors.toSet());
     }<% } %>
 }

--- a/generators/server/templates/src/main/java/package/web/rest/mapper/_UserMapper.java
+++ b/generators/server/templates/src/main/java/package/web/rest/mapper/_UserMapper.java
@@ -41,7 +41,7 @@ public interface UserMapper {
         return user;
     }
 <% if (databaseType == 'sql' || databaseType == 'mongodb') { %>
-        default Set<String> stringsFromAuthorities (Set < Authority > authorities) {
+    default Set<String> stringsFromAuthorities (Set < Authority > authorities) {
         return authorities.stream().map(Authority::getName)
             .collect(Collectors.toSet());
     }

